src/minidump.rs

```diff
@@ -1581,11 +1586,14 @@ where
     /// [`MinidumpStream`][stream] trait, this method allows reading
     /// the stream data as a specific type.
     ///
+    /// Note that the lifetime of the returned stream is bound to the lifetime of the this
+    /// `Minidump` struct itself and not to the lifetime of the data backing this minidump.
+    /// This is a consequence of how this struct relies on [Deref] to access the data.
```

Comment:
It seems like we can fix this to actually do the sensible thing, right? We'd just have to add another named lifetime and use that for `&self`, but use `'a` (which should probably be renamed to `'mdmp` as you've done elsewhere) to correctly tie the `MinidumpStream`'s lifetime to the data. That will probably also require telling rustc about the relation between the two lifetimes.

Reply:
Whoops, this PR completely escaped me.  Apologies.

Don't you describe how the lifetimes were organised before this change?  And there the lifetime of the reference returned with deref is always the same lifetime as the Minidump struct itself has IIUC.  The Minidump struct does tie all these lifetimes together with the lifetime specifiers on the deref and phantomdata.